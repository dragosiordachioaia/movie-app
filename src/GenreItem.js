import React, { Component } from "react";

import Checkbox from "./Checkbox";

export default class GenreItem extends Component {
  constructor(props) {
    super(props);

    // this.displayStuff = this.displayStuff.bind(this);
  }

  displayStuff() {}

  render() {
    return (
      <div className="genre-item">
        <Checkbox
          checked={this.props.checked}
          onCheck={() => this.props.onCheck(this.props)}
          label={this.props.name}
        />
      </div>
    );
  }
}
