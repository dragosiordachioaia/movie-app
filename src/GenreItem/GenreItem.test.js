import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import GenreItem from "./GenreItem";

jest.mock("Checkbox/Checkbox", () => "Checkbox");

let component;
let instance;

beforeEach(() => {
  component = renderer.create(<GenreItem />);
  instance = component.getInstance();
});

it("renders without crashing", () => {
  instance.props = {
    checked: true,
    name: "Action",
  };
  const result = instance.render();
  expect(result).toMatchSnapshot();
});
