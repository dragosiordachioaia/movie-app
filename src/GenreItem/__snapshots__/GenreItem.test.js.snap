// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="genre-item"
>
  <Checkbox
    checked={true}
    label="Action"
    onCheck={[Function]}
  />
</div>
`;
